### PR TITLE
FS-4941 - Replace black, flake8 and isort with ruff

### PR DESCRIPTION
**Pull request by wjrm500**
### Ticket

https://mhclgdigital.atlassian.net/browse/FS-4941

### Description

Commits...
- Replace black, flake8 and isort with ruff in dependencies and pre-commit config
- Apply automated safe fixes
- Apply automated unsafe fixes
- Manually fix remaining issues highlighted by ruff

### CODE REVIEW

File: app/db/queries/fund.py [hunk 1]
```diff
@@ -14,13 +14,13 @@ def add_organisation(organisation: Organisation) -> Organisation:
 def add_fund(fund: Fund) -> Fund:
     db.session.add(fund)
     db.session.commit()
-    current_app.logger.info(f"Fund added with fund_id: '{fund.fund_id}.")
+    current_app.logger.info("Fund added with fund_id: '%s'.", fund.fund_id)
```
Comment (samuelhwilliams) [inline app/db/queries/fund.py:17]: We might head this way in the future, but for now the consistent approach used everywhere else is:

`current_app.logger.info("Fund added with fund_id: '{fund_id}'.", extra=dict(fund_id=fund.fund_id))`

Comment (wjrm500) [reply app/db/queries/fund.py:17]: Noted, will convert to this now

Comment (wjrm500) [reply app/db/queries/fund.py:17]: Is `current_app.logger.info("Fund added with fund_id: '{fund_id}'.", extra={"fund_id": fund.fund_id})` okay or is `dict` syntax preferred?

Comment (wjrm500) [reply app/db/queries/fund.py:17]: Ah ignore me it's very minor, I'll use `dict` for consistency

Comment (samuelhwilliams) [reply app/db/queries/fund.py:17]: either/or is totally fine but yeah, they most use `dict()`

**Review by samuelhwilliams — APPROVED** [review]
nice 🙏

**Pull request merged**